src/test/java/org/springframework/data/elasticsearch/junit/jupiter/SpringDataElasticsearchExtension.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.elasticsearch.junit.jupiter;
+
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This extension class check in the {@link #beforeAll(ExtensionContext)} call if there is already a Elasticsearch
+ * cluster connection defined in the root store. If no, the connection to the cluster is defined according to the
+ * configuration, starting a local node if necessary. The connection is stored and will be closed when the store is
+ * shutdown at the end of all tests.
+ *
+ * @author Peter-Josef Meisch
+ */
+public class SpringDataElasticsearchExtension implements BeforeAllCallback {
```

Comment:
It would make sense to implement also `ParameterResolver` here. This is, to give non-Spring tests the chance to connect to Elasticsearch using a client so tests can inject `ClusterConnection` into the test method or a setup method.

Furthermore, we need to address the integration with Spring. Given the ordering aspect, we might want to make this class also implement `ContextCustomizerFactory` to customize the Spring context before it bootstraps (requires `META-INF/spring.factories`). We could register a resolvable dependency with 

```java
	@Override
	public ContextCustomizer createContextCustomizer(Class<?> testClass, List<ContextConfigurationAttributes> configAttributes) {
		return this::customizeContext;
	}

	private void customizeContext(ConfigurableApplicationContext context, MergedContextConfiguration mergedConfig) {

		context.getBeanFactory()
				.registerResolvableDependency(ClusterConnectionInfo.class, ClusterConnection
						.getClusterConnectionInfo());
	}
```

so that configuration classes can declare a `@Autowired ClusterConnectionInfo` and so we connect both pieces. 

I wasn't able to achieve the same using a `TestExecutionListener` as the listener invocation seems too late for our purposes.

Reply:
implemented this; I just didn't see the last connecting step from my implementation yesterday and then context customizing ;-) Now Autowiring works and parameter resolution as well